cardano-cli/src/Cardano/CLI/EraBased/Run/CreateTestnetData.hs

```diff
@@ -603,18 +602,18 @@ updateOutputTemplate
           , sgProtocolParams
           }
   where
-    getCoinForDistribution :: Integer -> m Natural
-    getCoinForDistribution inputCoin = do
-      let value = inputCoin - subtrahendForTreasury
-      when (value < 0) $ throwError $ GenesisCmdNegativeInitialFunds value
-      pure $ fromInteger value
+    nonDelegCoin = getCoinForDistribution nonDelegCoinRaw
+    delegCoin = getCoinForDistribution delegCoinRaw
+
+    getCoinForDistribution :: Integer -> Natural
```

Comment:
Hm. I don't like these hidden calculations. It is always better to be explicit.

Reply:
@Jimbo4350> note that it was like this before in `create-staked`.